programs/survival/fakeplayer_inv_editor.sc

```diff
@@ -0,0 +1,106 @@
+__config()->{
+    'scope'->'global',
+    'stay_loaded'->true
+};
+create_datapack('invupd',{
+'readme.txt'->['this data pack is created by scarpet','please dont touch it'],
+'data'->{'chyx/advancements/xd.json'->
+
+{'rewards'->{'function'->'chyx:invupd'},
+  'criteria'-> {
+    'example'-> {
+      'trigger'-> 'minecraft:inventory_changed',
+      'conditions'-> {
+      }
+    }
+  }
+},
+
+
+
+'chyx/functions/invupd.mcfunction'->'script run signal_event(\'invupd\', null, player())\nadvancement revoke @s only chyx:xd'}});
+global_nope=nbt('{nope:nopeChYx'+rand(1)+'nope}');
+
+global_slotmap=[[-1,7],[-2,1],[-3,2],[-4,3],[-5,4],...map(range(9),[_,45+_]),...map(range(27),[9+_,18+_])];
+
+
+global_fakeplayersscreen={};
+global_quick_craft={};
+__on_player_interacts_with_entity(creativeplayer, fakeplayer, hand)->(
+    if(hand=='mainhand',0,return());
+    if(fakeplayer~'player_type'=='fake',0,return());
+    if(creativeplayer~'player_type'=='fake',return());
+    //if(creativeplayer~'gamemode'=='creative',0,return());
+    if(global_fakeplayersscreen:fakeplayer,return());
+
```

Comment:
This could all be one big if. And remember to negate conditions when you want to check the oposite of what you are checking. `if(mycondition, 0, do_somethight())` is way less clear than `if(!mycondition, do_something)`.
```suggestion
    if(
        hand != 'mainhand', return(),
        fakeplayer~'player_type' != 'fake', return(),
        creativeplayer~'player_type' == 'fake', return(),
        //creativeplayer~'gamemode' != 'creative', return(),
        global_fakeplayersscreen:fakeplayer, return()
    );
```